app/data_model/list_store.py

```diff
@@ -12,22 +13,92 @@ def random_string(length):
     return ''.join(random.choice(ascii_letters) for _ in range(length))
 
 
+class ListRepresentation:
+    def __init__(
+        self, items: Optional[List] = None, primary_person: Optional[str] = None
+    ):
+        self.items = items or []
+        self.primary_person = primary_person
+
+    def __eq__(self, other):
+        if not isinstance(other, ListRepresentation):
+            return NotImplemented
+        return self.items == other.items and self.primary_person == other.primary_person
+
+    def remove_item_id(self, to_remove):
+        """ Remove an item from the list.
+        If the list item does not exist, this will not raise an error.
+        """
+        try:
+            self.items.remove(to_remove)
+        except ValueError:
+            pass
+
+        if self.primary_person == to_remove:
+            self.primary_person = None
+
+    @classmethod
+    def from_dict(cls, input_dict: dict):
```

Comment:
Should probably use the `typing.Mapping` type here to be more generic in acceptance